cmd/allocator/main.go

```diff
@@ -107,6 +113,16 @@ func main() {
 		for {
 			select {
 			// watch for events
+			case event := <-watcherTLS.Events:
+				h.tlsMutex.Lock()
+				tlsCert, err := readTLSCert()
+				if err != nil {
+					logger.WithError(err).Infof("could load TLS cert; keeping old one")
+				} else {
+					h.tlsCert = tlsCert
```

Comment:
While we are here can we change `h.caCertPool = caCertPool` to be in an else clause as well?

Reply:
Done!